exp/addons/internal/controllers/clusterresourceset_scope_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	addonsv1 "sigs.k8s.io/cluster-api/exp/addons/api/v1beta1"
+)
+
+func TestReconcileStrategyScopeNeedsApply(t *testing.T) {
+	tests := []struct {
+		name  string
+		scope *reconcileStrategyScope
+		want  bool
+	}{
+		{
+			name: "not binding",
```

Comment:
"not bound"?

Reply:
Updated to `ResourceBinding` in all test cases for clarity